### ISSUE TITLE
Add ability to selectively import Vue controllers

### ISSUE THREAD

**Issue by tpomerhn** [magico13/ha-emporia-vue#262]
I currently have three Vue 2 controllers (two 16-port and one 8-port) for my home in the US, and a 16-port in a condo I have internationally. I am about to complete construction of a third dwelling shortly, which will have probably 3-4 Vues in it because of all the sub-panels. 

While I love this integration, and just started using it, it's a bit of a pain to have my two different HA instances for the first two dwellings have to "disable" all of the sensors for devices that are not at their respective site. I guess it technically doesn't "hurt" to have each HA instance grab all sensor data for all devices everywhere but that's a bit overkill, especially when we're talking about 6-7 Vue units across three HA instances.

I'd like a feature to somehow limit the import of sensor data to a particular controller - or, ideally, a list of controllers. Each controller has a unique serial number in the Emporia app, so I would assume it's technically possible to only retrieve sensor data for one (or, preferably, a list) of serial numbers... or, at a minimum, grab all sensor data and then "drop" data that doesn't match a serial number.

Looks like the API supports a field called "manufacturerDeviceId" which is the serial number (from https://emporia-connect.xyt.co.za), so maybe it's possible?

**Comment by DabblerIOT** [comment]
Yes would absolutely be possible.   Just curious why you wouldn't put each residence on its own emporia account?   Make a email for each property?   Seems to me each emporia instance is one instance per residence... not alot of multiple site or residence features (if any) on one instance.

Each controller / plug or whatever gets its own gid which is used for main query that looks like this..

/AppAPI?apiMethod=getDeviceListUsages&deviceGids=[296617,296746,279499,279505,285180,306354]&instant=2024-03-06T19:41:15.933Z&scale=1MIN&energyUnit=KilowattHours

(those are my gids for some vues and plugs)

You could probably hack up your core emporia-vue hacs install pretty easy to make this change yourself if you wanted.

**Comment by tpomerhn** [comment]
Interesting thought - if I can figure this out, maybe I'll see if I can submit my changes to your code base. I'll have to dig in a bit to see how the API interacts with Emporia and I'll admit, I only know enough python to be dangerous. :) So, any support that the core team for this project can supply would be appreciated.

While I understand possibly having multiple accounts, the issue is I don't know how Emporia's app would handle that... i.e. would I have to "sign out" and then "sign in" every time I wanted to view a different residence? That negates some of the utility of the app, which I actually find is quite nice for viewing my two current residences in a single pane of glass. 

Perhaps I could create individual accounts per residence, and then one "master" account that has all of the Vue's registered to it, then "share" only the Vue's that are for a particular residence to the individual accounts, and have HA join just the individual one that is for its residence... hmm... then I could have the app signed into the "master" account but still get the granularity I need for HA. I may try that; do you happen to know if "shared" Vue's show up in this integration? I'll let you know my results.

Thanks for your support of this project!

**Comment by tpomerhn** [comment]
SUCCESS! Thank you for your suggestion.

For anyone else wanting to do the same - create individual, "site," accounts for each residence. Share just the Vue(s) from a "master" email account with that "site" account and have the HA instance join just that account. You'll only see the specific Vue(s) shared.

A further note - in my case, I had initially joined with a "master" account, and then removed it, rebooted HA, and then joined using the "site" account - but it still showed everything from "master." I had to remove it again, delete the entire Emporia Vue plugin/repo from HACS, reboot HA (not restart - had to reboot the VM), re-add the repo, and join. Then I only saw the "site" Vue(s).

For now, I hope this resolves my issues. Thanks again, DabblerIOT!
